### DISCUSSION TITLE
Using non-rgb data as input

### DISCUSSION THREAD

**Discussion by s-m-t-c** [opengeos/segment-geospatial#52] [Q&A] [answered]
Hi, well done on creating this library!

I am trying to segment single band Sentinel 1 data.

I've tried passing in a numpy array as well as configuring my geotiff a few different ways (datatype, shape etc). I receive a range of different errors when running the step:

`mask = "segment.tif"
sam.generate(
  image, mask, batch=True, foreground=True, erosion_kernel=(3, 3), mask_multiplier=255
)`

including:

`RuntimeError: permute(sparse_coo): number of dimensions in the tensor input does not match the length of the desired ordering of dimensions i.e. input.dim() = 2 is not equal to len(dims) = 3`

I'm curious if anyone has had success using segment-geospatial with single band or SAR data? If I dig around in the library I can see a few lines of code that appear to assume multi-band data but I could be mistaken.

![s1axarrayexmaple](https://github.com/opengeos/segment-geospatial/assets/22090545/06a0eae3-a96b-46ab-987b-b49542b645f6)

![S1croperror](https://github.com/opengeos/segment-geospatial/assets/22090545/69a7d5cc-a2ee-478b-9fff-c9265d1ed1c5)

**Comment by giswqs** [comment, marked as answer]
See https://github.com/facebookresearch/segment-anything/issues/123
